docs/tutorial/object_decomp.md

```diff
@@ -1,14 +1,161 @@
+
+The `ci` formats use palettes, which are declared separately. The shape you give the palette does not matter, but to avoid overlap errors it needs to fit into a rectangle; choose the one you think looks best.
+
+If in doubt, look at completed objects in the repo, and if still in doubt, ask.
+
+## Tools
 
 We are very fortunate that several nice tools have been written recently that are excellent for documenting asset files:
 - [Z64Utils](https://github.com/Random06457/Z64Utils/releases), for looking at displaylists, textures they reference, the skeleton, animations, etc.
 - [Texture64](https://github.com/queueRAM/Texture64/releases), for looking at textures in all the common N64 formats (needed since Z64Utils cannot interpret textures not explicitly referenced in displaylists currently)
 
+## Building and investigative modding
+
+Thankfully it is not necessary to do a full make from clean to check that a particular object file has been decompiled successfully.
+
+- With a valid xml file, run `extract_assets.py -s` on its object.
+- `make`
+
+If you want to change a texture, for example to see precisely where it is used, the following steps ensure it will be used in the build
+
+- Change and save the texture
+- Touch the C file in the same directory (that includes it)
+- make
+- Ironically, ERROR 1 implies success (your new texture has changed the checksum).
+
+If you'd rather not have it tell you about the checksum, you can run `make COMPARE=0` instead.
+
+---
+
+To revert to the original texture, you can just run `extract_assets.py -s` on the object again.
+
+N.B. doing this will overwrite every custom texture, as will running `make setup`.
+
+## Example
+
+An example of decompiling a particular object is given [here](object_decomp_example.md).
+
+## Tips and Tricks
+
+### Fixing improper-looking textures
+
+For CI4 and CI8 textures, you might see improper-looking textures like so:
+
+![An improper-looking CI8 texture](images/broken_texture.png)
+
+The reason this happens is because ZAPD couldn't determine the TLUT for the texture, so it couldn't use the proper palette. To fix this, you can supply a `TlutOffset` to the texture like so:
+```xml
+<Texture Name="gGiantFaceEyeOpenTex" OutName="giant_face_eye_open" Format="ci8" Width="32" Height="64" Offset="0x5A80" TlutOffset="0x5380" />
+```
+
+### Understanding texture animations
+
+Texture animations are new to Majora's Mask, and they can be pretty tricky to understand. Luckily, there's some extensive documentation on how they're strucutred [here](https://github.com/zeldaret/mm/blob/master/tools/ZAPD/ZAPD/ZTextureAnimation.cpp). One useful thing to remember is that empty texture animations take the form of `00 00 00 06 00 00 00 00`. The process that automatically generated all the object XMLs sometimes failed to recognize this as an empty texture animation, so it puts it in various blobs or fails to account for it at all.
```

Comment:
```suggestion
Texture animations are new to Majora's Mask, and they can be pretty tricky to understand. Luckily, there's some extensive documentation on how they're structured [here](https://github.com/zeldaret/mm/blob/master/tools/ZAPD/ZAPD/ZTextureAnimation.cpp). One useful thing to remember is that empty texture animations take the form of `00 00 00 06 00 00 00 00`. The process that automatically generated all the object XMLs sometimes failed to recognize this as an empty texture animation, so it puts it in various blobs or fails to account for it at all.
```
I feel like that documentation should probably go somewhere more accessible at some point, although not sure where.